deepchem/feat/mol_graphs.py

```diff
@@ -267,14 +271,9 @@ def agglomerate_mols(mols, max_deg=10, min_deg=0):
     atoms_per_mol = [mol.get_num_atoms() for mol in mols]
 
     # Get atoms by degree
-    atoms_by_deg = [
-        mol.get_atoms_with_deg(deg)
-        for deg in range(min_deg, max_deg + 1)
-        for mol in mols
-    ]
-
-    # stack the atoms
-    all_atoms = np.vstack(atoms_by_deg)
+    atoms_by_deg = np.concatenate([x.atom_features for x in mols])
+    degree_vector = np.concatenate([x.degree_list for x in mols], axis=0)
+    all_atoms = atoms_by_deg[degree_vector.argsort(kind='mergesort')]
```

Comment:
mergesort is key here -- We need a stable sort which the default sort, quicksort is not.

Reply:
Have to add a test to ensure that people don't change this to a non-stable sort.

Reply:
Put a comment to that effect in the code so people won't be tempted to.